src/main/scala/tech/cryptonomic/conseil/tezos/FeeOperations.scala

```diff
@@ -37,18 +45,20 @@ object FeeOperations extends LazyLogging {
     * Calculates average fees for each operation kind and stores them into a fees table.
     * @return
     */
-  def processTezosAverageFees(): Try[Unit] = {
+  def processTezosAverageFees()(implicit ex: ExecutionContext): Future[Option[Int]] = {
```

Comment:
Is explicitly writing that this function uses an ExecutionContext just good practice for documentation, or is there another benefit? (Totally fine either way)

Reply:
It's necessary, not perfunctory, as it's required to call `andThen` on the `Future` result.

I'm passing this as a parameter because I've removed the

`scala.concurrent.ExecutionContext.Implicits.global`

import from the top.

I consider this best-practice, both to control execution context as a (possibly implicit) parameters and to avoid using the `global` one.

Reply:
Got it, thanks for the clarification! andThen also does seems like a better way to handle things

Reply:
To be totally honest I rarely see anyone else using it, everyone just sticks to onComplete usually.

Though the idea of `andThen` is to sequence additional side-effects operation after the future, without changing it for real. And there are many cases where this seems to apply fittingly